docs/source/checks/tabular/model_evaluation/plot_weak_segments_performance.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+"""
+Weak Segments Performance
+********************************
+
+This notebooks provides an overview for using and understanding the weak segment performance check.
+
+**Structure:**
+
+* `What is the purpose of the check? <#what-is-the-purpose-of-the-check>`__
+* `Generate data & model <#generate-data-model>`__
+* `Run the check <#run-the-check>`__
+* `Define a condition <#define-a-condition>`__
+
+What is the purpose of the check?
+=================================
+The check is designed to help you easily identify weak spots of your model and provide a deepdive analysis into
+its performance on different segments of your data. Specifically, it is designed to help you identify the model weakest
+segments in the data distribution for further improvement and visibility purposes.
+
+In order to achieve this, the check trains several simple tree based models which try to predict the error of the
+user provided model on the dataset. The relevant segments are detected by analyzing the different
+leafs of the trained trees.
```

Comment:
Would add here another subtitle... (e.g. like here): https://docs.deepchecks.com/stable/checks_gallery/tabular/train_test_validation/plot_whole_dataset_drift.html

and maybe explain a bit more (how we detect and then split around) for the curious data scientist